src/dpop.js

```diff
@@ -1,6 +1,39 @@
 /* @flow */
 /* eslint-disable promise/no-native, no-restricted-globals */
 
+type KeyPair = {|
+  privateKey: mixed,
+  publicKey: mixed,
+|};
+
+type GenerateKeyPair = () => Promise<KeyPair>;
+
+// https://datatracker.ietf.org/doc/html/rfc7518#section-3.1
+const KEY_OPTIONS = {
+  create: {
+    name: "ECDSA",
+    namedCurve: "P-256",
+  },
+  extractable: false,
+  usages: ["sign", "verify"],
+};
+
+let keyPair;
+export const generateKeyPair: GenerateKeyPair = async () => {
+  if (!keyPair) {
+    const { create, extractable, usages } = KEY_OPTIONS;
+    const { publicKey, privateKey } = await window.crypto.subtle.generateKey(
+      create,
+      extractable,
+      usages
+    );
+
+    keyPair = keyPair || { publicKey, privateKey };
```

Comment:
^ this is for https://eslint.org/docs/latest/rules/require-atomic-updates 

we know no other function mutates `keyPair` but eslint wanted me to read the value one last time before reassignment.